db/seeds/base/add_schedules.rb

```diff
@@ -0,0 +1,39 @@
+require "csv"
+
+class AddSchedules
+  attr_reader :path_to_csv
+
+  def initialize(path_to_csv:)
+    @path_to_csv = path_to_csv
+  end
+
+  def call
+    rows.each do |row|
+      course_group = CourseGroup.find_by!(name: row["course-group-name"])
+      cohort = Cohort.find_by!(start_year: row["schedule-cohort-year"].to_i)
+
+      schedule = Schedule.find_or_initialize_by(
+        identifier: row["schedule-identifier"],
+        cohort:,
+      )
+
+      schedule.course_group = course_group
+      schedule.name = row["schedule-name"]
+      schedule.applies_from = row["schedule-applies-from"]
+      schedule.applies_to = row["schedule-applies-to"]
+      schedule.allowed_declaration_types = row["declaration-types"].split("|")
+
+      schedule.save!
+    end
+  end
+
+  def rows
+    @rows ||= CSV.read(
```

Comment:
Is it possible to avoid using CSVs in NPQ reg? We're hoping to move away from this setup and do most cohort set up in the admin, so I was thinking is there an easier way we can create schedules using factories instead? I think we did something similar in statements?